frontend/src/store.ts

```diff
@@ -1,18 +1,62 @@
-import { configureStore } from '@reduxjs/toolkit'
+import { ThunkMiddleware, configureStore } from '@reduxjs/toolkit'
 import { createLogger } from 'redux-logger'
 import rootReducer from './reducers'
+import { showNotification } from './modules/notification/actions'
+import { logOut } from './modules/auth/actions'
+import shouldIgnoreError from './utils/shouldIgnoreError'
 
 const logger = createLogger({
 	level: 'info'
 })
 
+const notificationError: ThunkMiddleware = ({ dispatch }) => next => action => {
+	const TOKEN_EXPIRED_MESSAGE = 'Given token not valid for any token type'
+
+	function getErrorDescription(error: any) {
+		/* HTTP errors are handled specially because they include the URL
+		 * in the error message and we don't want many very similar errors
+		 * to show up, just show one.
+		 */
+		if (typeof error.status === 'number')
+			return {
+				message: `HTTP Error ${error.status}: ${error.statusText}`,
+				details: error.url,
+			}
+
+		return {
+			message: error.message,
+			details: error.stack,
+		}
+	}
+
+	if (!action.error)
+		return next(action)
+
+	if (action.error.message.includes(TOKEN_EXPIRED_MESSAGE)) {
+		dispatch(logOut())
+		return next(action)
+	}
+
+	if (shouldIgnoreError(action))
+		return next(action)
+
+	const error = getErrorDescription(action.error)
+	dispatch(showNotification(error.message, error.details))
```

Comment:
Might be better to just use `notify` here to keep it clear what is being used for an id, and that an error alert is being popped.